### DISCUSSION TITLE
Unexpected error with use_future, use_effect hooks

### DISCUSSION THREAD

**Discussion by onweru** [DioxusLabs/dioxus#1956] [Q&A] [answered]
I know this might be too soon but maybe there's someone who can guide me.  If not, I will wait for `v0.5` release.

I'm trying out the changes in the main branch, mainly to test #1791. I'm using the web launcher and so far most things work well. Some hooks e.g use_effect, use_future throw errors. checking through the source and [examples](https://github.com/DioxusLabs/dioxus/blob/c0f2e8391cf0be205b3cbee16deb1c25fe4f7f2e/examples/backgrounded_futures.rs#L49-L54), it appears that __scope__ and __deps__ are no longer needed as arguments. But I still get `expected 3 arguments found 1` when I try something like:

```
use_future(move || async move {
    get_items(endpoint).await.json::<T>().await
})
``` 
I'm curious if my Cargo.toml file 👇🏼  is the issue or some other things I missed.

```
dioxus = { git = "https://github.com/DioxusLabs/dioxus", rev = "3fef42a43bebcdfbce8717b8c99915f93403c4d8" }
dioxus-router = { git = "http://github.com/DioxusLabs/dioxus", rev = "3fef42a43bebcdfbce8717b8c99915f93403c4d8", features = ["web"] }
dioxus-web = { git = "https://github.com/DioxusLabs/dioxus", rev = "3fef42a43bebcdfbce8717b8c99915f93403c4d8" }
dioxus-html = { git = "http://github.com/DioxusLabs/dioxus", rev = "3fef42a43bebcdfbce8717b8c99915f93403c4d8" }
```

**Comment by ealmloff** [comment, marked as answer]
The code you posted looks mostly correct. You can use `use_resource` instead of `use_future` if you want to return a value from the future:
```rust
let resource = use_resource(move || async move {
    get_items(endpoint).await.json::<T>().await
});
```

If you are using rust analyzer, you might need to restart your editor to get an updated error after changing the version. 

Some other 0.5 resources/changes:
The docs for 0.5 are updated [here](https://github.com/DioxusLabs/docsite/pull/202). They include a [migration guide](https://github.com/ealmloff/docsite/blob/dioxus-next/docs-src/0.4/en/migration/hooks.md#dependencies) to help you update from 0.4 -> 0.5 which might be useful.

In 0.5, you also don't need a separate dependency for the renderer, router and html crates:
```toml
dioxus = { git = "https://github.com/DioxusLabs/dioxus", rev = "3fef42a43bebcdfbce8717b8c99915f93403c4d8", features = ["web", "router"] }
```
```rust
use dioxus::prelude::*;

fn main() {
    launch(|| rsx! { Router<Route> {} })
}
```

**Reply by onweru** [reply]
Thank you, this has been very helpful. All good 🙇🏼
